libs/community/langchain_community/document_loaders/blob_loaders/cloud_blob_loader.py

```diff
@@ -0,0 +1,290 @@
+            return self.data
+        elif isinstance(self.data, str):
+            return self.data.encode(self.encoding)
+        elif self.data is None and self.path:
+            return AnyPath(self.path).read_bytes()  # type: ignore
+        else:
+            raise ValueError(f"Unable to get bytes for blob {self}")
+
+    @contextlib.contextmanager
+    def as_bytes_io(self) -> Generator[Union[BytesIO, BufferedReader], None, None]:
+        """Read data as a byte stream."""
+        from cloudpathlib import AnyPath
+
+        if isinstance(self.data, bytes):
+            yield BytesIO(self.data)
+        elif self.data is None and self.path:
+            return AnyPath(self.path).read_bytes()  # type: ignore
+        else:
+            raise NotImplementedError(f"Unable to convert blob {self}")
+
+
+def _url_to_filename(url: str) -> str:
+    """
+    Convert file:, s3:, az: or gs: url to localfile.
+    If the file is not here, download it in a temporary file.
+    """
+    from cloudpathlib import AnyPath
+
+    url_parsed = urlparse(url)
+    suffix = Path(url_parsed.path).suffix
+    if url_parsed.scheme in ["s3", "az", "gs"]:
+        with AnyPath(url).open("rb") as f:  # type: ignore
+            temp_file = tempfile.NamedTemporaryFile(suffix=suffix, delete=False)
+            while True:
+                buf = f.read()
+                if not buf:
+                    break
+                temp_file.write(buf)
+            temp_file.close()
+            file_path = temp_file.name
+    elif url_parsed.scheme in ["file", ""]:
+        file_path = url_parsed.path
+    else:
+        raise ValueError(f"Scheme {url_parsed.scheme} not supported")
+    return file_path
+
+
+def _make_iterator(
+    length_func: Callable[[], int], show_progress: bool = False
+) -> Callable[[Iterable[T]], Iterator[T]]:
+    """Create a function that optionally wraps an iterable in tqdm."""
+    if show_progress:
+        try:
+            from tqdm.auto import tqdm
+        except ImportError:
+            raise ImportError(
+                "You must install tqdm to use show_progress=True."
+                "You can install tqdm with `pip install tqdm`."
+            )
+
+        # Make sure to provide `total` here so that tqdm can show
+        # a progress bar that takes into account the total number of files.
+        def _with_tqdm(iterable: Iterable[T]) -> Iterator[T]:
+            """Wrap an iterable in a tqdm progress bar."""
+            return tqdm(iterable, total=length_func())
+
+        iterator = _with_tqdm
+    else:
+        iterator = iter  # type: ignore
+
+    return iterator
+
+
+# PUBLIC API
+
+
+class CloudBlobLoader(BlobLoader):
+    """Load blobs from cloud URL or file:.
+
+    Example:
+
+    .. code-block:: python
+
+        loader = CloudBlobLoader("s3://mybucket/id")
+        for blob in loader.yield_blobs():
+            print(blob)
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        url: Union[str, "AnyPath"],
```

Comment:
@pprados you probably already know this, but if not

you can use 

`from __future__ import annotations`

And I think that will allow you to not worry about the `"` marks around types that are imported inside the TYPE_CHECKING if conditional block